pkgs/by-name/du/duckdb/package.nix

```diff
@@ -1,56 +1,83 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, git
 , cmake
 , ninja
 , openssl
-, openjdk11
 , python3
 , unixODBC
-, withJdbc ? false
 , withOdbc ? false
 }:
 
 let
-  enableFeature = yes: if yes then "ON" else "OFF";
-  versions = lib.importJSON ./versions.json;
+  canExecute = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+  # needs to match source/src/include/duckdb/common/platform.h
+  duckDBPlatform = let
+    os = if stdenv.hostPlatform.parsed.kernel.name == "darwin"
+      then "osx"
+      else stdenv.hostPlatform.parsed.kernel.name;
+    arch = {
+      "aarch64" = "arm64";
+      "x86_64" = "amd64";
+    }.${stdenv.hostPlatform.parsed.cpu.name} or stdenv.hostPlatform.parsed.cpu.name;
+  in
+    "${os}_${arch}";
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "duckdb";
-  inherit (versions) rev version;
+  version = "1.0.0";
 
   src = fetchFromGitHub {
-    # to update run:
-    # nix-shell maintainers/scripts/update.nix --argstr path duckdb
-    inherit (versions) hash;
     owner = "duckdb";
     repo = "duckdb";
     rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-6pP0MZs4IClszNcLTN+nkcuKml1rIG4FY/ybRV6nINU=";
+
+    # pull the git revision from the tar file and store it in .git/HEAD
+    nativeBuildInputs = [ git ];
+    # Ignore broken pipe error from gzip as git get-tar-common-id exits after
+    # reading 1k from the input stream.
+    postFetch = ''
+      mkdir "$out"/.git
+      (gzip -cd "$renamed" 2>/dev/null || true) | \
+        git get-tar-commit-id > "$out"/.git/HEAD
+    '';
   };
 
   outputs = [ "out" "lib" "dev" ];
 
   nativeBuildInputs = [ cmake ninja python3 ];
   buildInputs = [ openssl ]
-    ++ lib.optionals withJdbc [ openjdk11 ]
     ++ lib.optionals withOdbc [ unixODBC ];
 
   cmakeFlags = [
-    "-DDUCKDB_EXTENSION_CONFIGS=${finalAttrs.src}/.github/config/in_tree_extensions.cmake"
-    "-DBUILD_ODBC_DRIVER=${enableFeature withOdbc}"
-    "-DJDBC_DRIVER=${enableFeature withJdbc}"
-    "-DOVERRIDE_GIT_DESCRIBE=v${finalAttrs.version}-0-g${finalAttrs.rev}"
-  ] ++ lib.optionals finalAttrs.doInstallCheck [
-    # development settings
-    "-DBUILD_UNITTESTS=ON"
+    (lib.cmakeBool "BUILD_ODBC_DRIVER" withOdbc)
+    (lib.cmakeBool "BUILD_UNITTESTS" finalAttrs.doInstallCheck)
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    "-DCMAKE_OSX_DEPLOYMENT_TARGET=${stdenv.hostPlatform.darwinMinVersion}"
+  ]
+  ++ lib.optionals (!canExecute) [
+    "-DDUCKDB_EXPLICIT_PLATFORM=${duckDBPlatform}"
   ];
 
-  postInstall = ''
-    mkdir -p $lib
-    mv $out/lib $lib
+  preConfigure = ''
+    cmakeFlagsArray+=( "-DDUCKDB_EXTENSION_CONFIGS=$PWD/.github/config/in_tree_extensions.cmake" )
+    cmakeFlagsArray+=( "-DINSTALL_BIN_DIR=$out/bin" )
+    cmakeFlagsArray+=( "-DINSTALL_INCLUDE_DIR=$dev/include" )
+    cmakeFlagsArray+=( "-DINSTALL_LIB_DIR=$lib/lib" )
```

Comment:
You can use ``placeholder "out"`` to get those variables in nix code and then these can be moved to cmakeFlags

Reply:
thx. i opted not to use `placeholder "xxx"` because i had to use `cmakeFlagsArray` already for a different flag and using env variables means that the build can run locally under nix-shell if out, dev and lib are re-defined to a writable location whereas `placeholcer` will always point to the nix store and break local builds.

Reply:
FWIW i just opened a PR upstream to use gnu install dirs so hopefully this code can get removed in the next release.
https://github.com/duckdb/duckdb/pull/12509